velox/expression/CastExpr.cpp

```diff
@@ -554,6 +554,14 @@ void CastExpr::applyPeeled(
     result = castFromDate(rows, input, context, toType);
   } else if (toType->isDate()) {
     result = castToDate(rows, input, context, fromType);
+  } else if (fromType->isUnKnown()) {
+    // Presto fail() function return type is UNKNOWN.
```

Comment:
fail function raises an error, not returning null.

Reply:
The fail return type signature is unknown. But, yes, it always raises an error.

Reply:
Please, remove this comment.

Reply:
Removed.